lib/web_ui/dev/test_runner.dart

```diff
@@ -850,3 +876,23 @@ class TestBuildInput {
 }
 
 class TestFailureException implements Exception {}
+
+/// Prints the name of the test, but suppresses all other test output.
+///
+/// This is useful to prevent pollution of logs by tests that are expected to
+/// fail.
+class NameOnlyReporter implements hack.Reporter {
+  NameOnlyReporter(hack.Engine testEngine) {
+    testEngine.onTestStarted.listen(_printTestName);
```

Comment:
I'm not sure if we need this, but we probably do:
```suggestion
    final sub = testEngine.onTestStarted.listen(_printTestName);
    testEngine.success.then((_) {
      sub.cancel();
    });
```